xwiki-rendering-macros/xwiki-rendering-macro-figure/src/main/java/org/xwiki/rendering/internal/macro/figure/FigureCaptionMacro.java

```diff
@@ -86,13 +87,16 @@ public boolean supportsInlineMode()
     {
         List<Block> result = Collections.emptyList();
 
-        // If we're not inside a FigureBlock then don't do anything.
-        Block parent = context.getCurrentMacroBlock().getParent();
-        if (parent instanceof FigureBlock) {
+        Block parentBlock = context.getCurrentMacroBlock().getParent();
+
+        // we don't want to do anything if we're not in a FigureBlock
+        if (parentBlock instanceof FigureBlock) {
             XDOM xdom = this.contentParser.parse(content, context, false, false);
             List<Block> figureCaptionChildren = xdom.getChildren();
             this.parserUtils.removeTopLevelParagraph(figureCaptionChildren);
-            result = Collections.singletonList(new FigureCaptionBlock(figureCaptionChildren));
+            List<Block> contentBlock = Collections.singletonList(new FigureCaptionBlock(figureCaptionChildren));
+
+            result = Collections.singletonList(new MetaDataBlock(contentBlock, this.getUnchangedContentMetaData()));
```

Comment:
Small detail: there's no need to prefix with "this.". We do that only for fields normally.